src/strategies/seedify-cumulative-voting-power-hodl-staking-farming/index.ts

```diff
@@ -0,0 +1,199 @@
+];
+
+const getStakingBalanceOf = (stakedBalances: any, userIndex: any) => {
+  let sum: number = 0;
+  let balance: any;
+  for (
+    let stakingContractIndex = 0;
+    stakingContractIndex < stakedBalances.length;
+    stakingContractIndex++
+  ) {
+    balance = toDecimals(stakedBalances[stakingContractIndex][userIndex]['0']);
+    sum += balance;
+  }
+  return sum;
+};
+
+const toDecimals = (bigNumber: any) => {
+  return parseFloat(formatUnits(bigNumber.toString(), 18));
+};
+
+const calculateBep20InLPForUser = (
+  lpStaked: any,
+  totalLPSupply: any,
+  totalBep20InPool: any
+) => {
+  lpStaked = toDecimals(lpStaked['0']);
+
+  return (lpStaked / totalLPSupply) * totalBep20InPool;
+};
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const createStakingPromises: any = (stakingAddresses: any[]) => {
+    const promises: any = [];
+    for (let i = 0; i < stakingAddresses.length; i++) {
+      promises.push(
+        createPromise(sfundStakingAbi, stakingAddresses[i], 'userDeposits')
+      );
+    }
+    return promises;
+  };
+
+  const createPromise: any = (
+    abi: any,
+    contractAddress: any,
+    functionToCall: any
+  ) => {
+    return multicall(
+      network,
+      provider,
+      abi,
+      addresses.map((address: any) => [
+        contractAddress,
+        functionToCall,
+        [address]
+      ]),
+      { blockTag }
+    );
+  };
+
+  // required to use: erc20BalanceOfStrategy
+  options.address = options.sfundAddress;
+
+  //////// return SFUND, in user's wallet ////////
+  let score: any = erc20BalanceOfStrategy(
+    space,
+    network,
+    provider,
+    addresses,
+    options,
+    snapshot
+  );
+
+  //////// return LP from SFUND-BNB pool, deposited into farming contract ////////
+  // current farming
+  let userLPStaked_SFUND_BNB: any = createPromise(
+    farmingAbi,
+    options.farmingAddress_SFUND_BNB,
+    'userDeposits'
+  );
+  // legacy farming
+  let userLPStaked_SFUND_BNB_legacyFarming: any = createPromise(
+    farmingAbi,
+    options.legacyfarmingAddress_SFUND_BNB,
+    'userDeposits'
+  );
+
+  //////// return LP from SNFTS-SFUND pool, deposited into farming contract ////////
+  let userLPStaked_SNFTS_SFUND: any = createPromise(
+    farmingAbi,
+    options.farmingAddress_SNFTS_SFUND,
+    'userDeposits'
+  );
+
+  //////// return user's SFUND balance in staking contract (IDOLocking) ////////
+  let sfundStaking: any = createStakingPromises(options.sfundStakingAddresses);
+  let legacySfundStaking: any = createStakingPromises(
+    options.legacySfundStakingAddresses
+  );
+
+  const result = await Promise.all([
+    score,
+    userLPStaked_SFUND_BNB,
+    userLPStaked_SFUND_BNB_legacyFarming,
+    userLPStaked_SNFTS_SFUND,
+    ...sfundStaking,
+    ...legacySfundStaking
+  ]);
```

Comment:
We still send many requests 🤔 make sure to call only 5 https://github.com/snapshot-labs/snapshot-strategies#code

Reply:
Hi @ChaituVR is that alright if we make 5 calls multiple times?, e.g.
```
const result = await Promise.all([
    score,
    userLPStaked_SFUND_BNB,
    userLPStaked_SFUND_BNB_legacyFarming,
    userLPStaked_SNFTS_SFUND
]); 

const result2 = await Promise.all([
    score,
    userLPStaked_SFUND_BNB,
    userLPStaked_SFUND_BNB_legacyFarming,
    userLPStaked_SNFTS_SFUND
]); 
```

Reply:
Hello @Theo6890 No 😄 we have limited resources, in case if you want to call too many, you can try to use `api-v2` strategy

Reply:
[api-v2](https://github.com/snapshot-labs/snapshot-strategies/tree/master/src/strategies/api-v2)

Reply:
If run on limited ressources, would it work to break this strategy into multiple strategies and then combine them on the dApp?

Reply:
No sorry for not being clear, i mean running your own server that can return voting power, and use `api-v2` strategy for it